internal/pkg/flags/types.go

```diff
@@ -1,9 +1,18 @@
 package flags
 
+import "sigs.k8s.io/controller-runtime/pkg/client"
+
 type ManagementCluster struct {
 	BaseDomain string
 	Name       string
 	Namespace  string
 	Provider   string
 	Region     string
 }
+
+func ToObjectKey(name string, namespace string) client.ObjectKey {
```

Comment:
Maybe you can have a ToObjectKey method instead to not have to pass the name and namespace?
```suggestion
func (m ManagementCluster) ToObjectKey() client.ObjectKey {
```